src/TwigComponent/doc/index.rst

```diff
@@ -1058,6 +1058,141 @@ Exclude specific attributes:
       My Component!
     </div>
 
+Component with Complex Variants (CVA)
+-------------------------------------
+
+CVA (Class Variant Authority) is a concept from the JS world (https://cva.style/docs/getting-started/variants).
+It's a concept used by the famous shadcn/ui library (https://ui.shadcn.com).
+CVA allows you to display a component with different variants (color, size, etc.),
+to create highly reusable and customizable components.
+You can use the cva function to define variants for your component.
+The first argument of the cva function is the base classes, the classes present in all variants. Then the second
+argument is an array of variants.
+
+.. code-block:: html+twig
+
+    {# templates/components/Alert.html.twig #}
+    {% props color = 'blue', size = 'md' %}
+
+     {% set alert = cva('alert ' ~ attributes.render('class'), {
+        color: {
+            blue: 'bg-blue',
+            red: 'bg-red',
+            green: 'bg-green',
+        },
+        size: {
+            sm: 'text-sm',
+            md: 'text-md',
+            lg: 'text-lg',
+        }
+    }) %}
```

Comment:
```suggestion
     {% set alert = cva({
        base: ['alert', attributes.render('class')], {
        color: {
            blue: 'bg-blue',
            red: 'bg-red',
            green: 'bg-green',
        },
        size: {
            sm: 'text-sm',
            md: 'text-md',
            lg: 'text-lg',
        }
    }) %}
```

Reply:
I made the change so the base now accepts array. But I think is easier if you just use the function apply to append your class.
`cva.apply(cva, attributes.render('class'))`

Reply:
Yes, I agree, but would base be the place to prepend classes if that's your intention?

Reply:
Yes this why I added the ability to have list for base, I think your right it can be useful in some situation

Reply:
But anyway, this matches [cva.style](https://cva.style/docs/api-reference#parameters)